src/utils/response/rawApiResponse.ts

```diff
@@ -18,22 +20,52 @@ export type RawApiResponse = {
   ErrorMessage?: string;
   ErrorInternal?: string;
   ErrorExtended?: unknown;
+  _ignoreTypeGuard?: boolean; // workaround until SP-351 is done, we don't want to update every test by hand
 };
 
-export async function createRawApiResponse(promise: Promise<AxiosResponse>): Promise<RawApiResponse> {
-  const axiosResponse: AxiosResponse = await promise;
-
-  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-  if (axiosResponse === undefined) {
-    throw new Error('The Api response is unsuccessful');
+function handleResponse(axiosResponse: AxiosResponse): RawApiResponse {
+  if (isRawApiResponse(axiosResponse.data) && axiosResponse.data.Success) {
+    axiosResponse.data._ignoreTypeGuard = undefined;
+    return axiosResponse.data;
   }
+  throw axiosResponse;
+}
 
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-  const rawApiResponse: RawApiResponse = axiosResponse.data;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function handleError(error: any) {
+  const apiResponseError: TimeTacApiError = {
+    reason: ErrorReason.ReponseFailed,
+    _plainError: JSON.stringify(error),
+    response: undefined,
+  };
 
-  if (!rawApiResponse.Success) {
-    throw new Error('The Api response is unsuccessful');
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  if (error.data !== undefined) {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    apiResponseError.response = error.data;
+  }
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  if (Boolean(error.response) && Boolean(error.response.data)) {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    apiResponseError.response = error.response.data;
   }
 
-  return rawApiResponse;
+  return Promise.reject(apiResponseError);
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function isRawApiResponse(response: any): response is RawApiResponse {
+  const hasHost = 'Host' in response;
+  const hasCodeversion = 'Codeversion' in response;
+  const hasSuccess = 'Success' in response;
+  const hasSuccessNested = 'SuccessNested' in response;
+  const hasRequestStartTime = 'RequestStartTime' in response;
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  const hasIgnoreTypeGuard = '_ignoreTypeGuard' in response && response._ignoreTypeGuard === true;
+
+  return hasIgnoreTypeGuard || (hasHost && hasCodeversion && hasSuccess && hasSuccessNested && hasRequestStartTime);
+}
```

Comment:
Here just replacing `any` with `Record<string, unknown>` works
```suggestion
function isRawApiResponse(response: Record<string, unknown>): response is RawApiResponse {
  const hasHost = 'Host' in response;
  const hasCodeversion = 'Codeversion' in response;
  const hasSuccess = 'Success' in response;
  const hasSuccessNested = 'SuccessNested' in response;
  const hasRequestStartTime = 'RequestStartTime' in response;
  const hasIgnoreTypeGuard = '_ignoreTypeGuard' in response && response._ignoreTypeGuard === true;

  return hasIgnoreTypeGuard || (hasHost && hasCodeversion && hasSuccess && hasSuccessNested && hasRequestStartTime);
}
```